src/jobs/models/jobManager.js

```diff
@@ -65,18 +65,21 @@ module.exports.createJob = async (job) => {
         if (job.run_immediately) {
             const latestDockerImage = await dockerHubConnector.getMostRecentRunnerTag();
             const test = await testsManager.getTest(job.test_id);
-            const report = await createReportForJob(test, insertedJob);
-            reportId = report.report_id;
+            report = await createReportForJob(test, insertedJob);
             const jobSpecificPlatformRequest = await createJobRequest(jobId, report.report_id, insertedJob, latestDockerImage, configData, contextId);
             await jobConnector.runJob(jobSpecificPlatformRequest, job);
         }
         if (job.cron_expression) {
             addCron(jobId, insertedJob, job.cron_expression, configData);
         }
         logger.info('Job deployed successfully');
-        return createResponse(jobId, job, reportId);
+        return createResponse(jobId, job, report.report_id);
     } catch (error) {
         logger.error(error, 'Error occurred trying to create new job');
+        if (report) {
+            logger.info('Removing the failed job report');
```

Comment:
logger.info(`Changing report: ${report.report_id} to failed after it failed to start`);